### PR TITLE
GH-43869: [Java][CI] Flight related failure in the AMD64 Windows Server 2022 Java JDK 11 CI

### PR DESCRIPTION
**Pull request by vibhatha**
### Rationale for this change

CIs have been consistently failing on windows recently due to an issue with derby configuration. This PR investigates a solution for this. 

### What changes are included in this PR?

Changing the flow of the exception handling and state return. 

### Are these changes tested?

Via existing test cases. 

### Are there any user-facing changes?

No
* GitHub Issue: #43869

### CODE REVIEW

**Comment by vibhatha** [comment]
@gituhb-actions crossbow submit -g java

**Comment by danepitkin** [comment]
Hmm, do we know when this first started failing?

**Comment by vibhatha** [comment]
> Hmm, do we know when this first started failing?

I haven't tracked that yet, @danepitkin I was looking into crossbows...

**Comment by vibhatha** [comment]
@danepitkin as it seems this started to occur 2 weeks ago, so roughly after Aug 14th. I haven't yet recognized a PR that does this, even the downgrade doesn't work, so it should be something internally changed and we have to reflect that in the way we have used the APIs.

**Comment by vibhatha** [comment]
@github-actions crossbow submit -g java

File: java/flight/flight-sql/src/test/java/org/apache/arrow/flight/sql/example/FlightSqlExample.java [hunk 1]
```diff
@@ -253,36 +252,36 @@ public FlightSqlExample(final Location location, final String dbName) {
...
-    boolean wasSuccess;
     final Path path = Paths.get("target" + File.separator + dbName);
 
     try (final Stream<Path> walk = Files.walk(path)) {
       /*
        * Iterate over all paths to delete, mapping each path to the outcome of its own
-       * deletion as a boolean representing whether or not each individual operation was
+       * deletion as a boolean representing whether each individual operation was
        * successful; then reduce all booleans into a single answer, and store that into
        * `wasSuccess`, which will later be returned by this method.
```
Comment (lidavidm) [inline java/flight/flight-sql/src/test/java/org/apache/arrow/flight/sql/example/FlightSqlExample.java:262]: wasSuccess doesn't exist anymore, let's update the comment

Comment (vibhatha) [reply java/flight/flight-sql/src/test/java/org/apache/arrow/flight/sql/example/FlightSqlExample.java:262]: Sorry missed that.

**Comment by vibhatha** [comment]
@lidavidm updated the PR.

**Review by lidavidm — APPROVED** [review]

**Pull request merged**